shell/browser/api/electron_api_browser_window.cc

```diff
@@ -304,8 +304,9 @@ void BrowserWindow::OnWindowIsKeyChanged(bool is_key) {
 
 void BrowserWindow::OnWindowResize() {
 #if defined(OS_MAC)
-  if (!draggable_regions_.empty())
-    UpdateDraggableRegions(draggable_regions_);
+  // Always update draggable regions here to ensure draggable bounds
+  // are recalculated for BrowserViews if any exist.
+  UpdateDraggableRegions(draggable_regions_);
```

Comment:
If there is a draggable region to update on resize then it would be signalled by `RenderFrameObserver::DraggableRegionsChanged` and the changes should get propagated via `ExtendedWebContentsObserver::OnDraggableRegionsUpdated`. Why should this update `Browserview` regions for empty  `draggable_regions_` ?

Reply:
@deepak1556 when draggable regions are empty on a frameless window, we create one giant `NSView` that is draggable for associated `BrowserViews`, and then on top of that add "exclusion zones", which for frameless windows with _no draggable regions_ will be the full size of the BrowserWindow. If we resize the window, what that means is that the BrowserWindow will resize, but the `NSView` we created _won't_ be resized, so we'll end up with incorrectly resizable _parts_ of the window.

Reply:
In that case can we just call to update the draggable regions of browser view here. I am looking to short-circuit the situation where there is no browser view and there are empty draggable regions for window.